analytics.go

```diff
@@ -65,22 +66,17 @@ func (a *AnalyticsRecord) GetGeo(ipStr string) {
 	if !config.AnalyticsConfig.EnableGeoIP {
 		return
 	}
-
 	// Not great, tightly coupled
 	if analytics.GeoIPDB == nil {
 		return
 	}
 
-	// Sometimes it is empty, we can't look up mpty IP addresses
-	if ipStr == "" {
+	record, err := geoIPLookup(ipStr)
+	if err != nil {
+		log.Warn("GeoIP Failure (not recorded): ", err)
```

Comment:
Just out curiosity, can we temporary make it Error, and add "ipStr" to logging info, to see if number errors is changed, and if our assumption about Ipv6 is right. Maybe there are more `ipStr` issues we do not aware about?

Overall looks good.